src/containers/Main/AuthenticatedWrapper.tsx

```diff
@@ -4,25 +4,35 @@ import { useRouter } from '../../utils/hooks/useRouter'
 import isLoggedIn from '../../utils/helpers/loginCheck'
 import NonRegisteredLogin from '../User/NonRegisteredLogin'
 import { Redirect } from 'react-router'
-import { Loading } from '../../components'
 
 const AuthenticatedContent: React.FC = () => {
-  const {
-    location,
-    query: { sessionId },
-  } = useRouter()
-  // Location is not fully loaded unless 'state' is present, TODO try upgrading react router
-  // without this sessionId is missing when navigating via Click Here in forgot password verification
-  if ('state' in location) {
```

Comment:
This was how @andreievg tried to fix the problem initially. However, it doesn't work fully, as the "query" (and "state") props never refresh, so they never actually get a value.

Reply:
I was contemplating manually parsing location too